src/routes/chatRoutes.ts

```diff
@@ -0,0 +1,25 @@
+import { Server,Socket } from "socket.io";
+import { Router } from "express";
+import { sendMessages,getPastMessages,joiRoom } from "../controllers/chatsController";
+ const chatRoutes = (io: Server): void => {
+    let connectedClients: Set<string> = new Set();
+    io.on('connection', async(socket: Socket) => {
+        connectedClients.add(socket.id);
+        io.emit('connected client', connectedClients.size);
+        await getPastMessages(socket);
+
+        socket.on('disconnect', () => {
+            io.emit('removed');
+            connectedClients.delete(socket.id);
+            io.emit('dis connected client', connectedClients.size);
+        });
+
+        socket.on('chat message', async(data: any) => {
+            await sendMessages(io, data);
+        });
+        
+    });
+}
+export const joinChatRoomRoutes = Router();
+joinChatRoomRoutes.get('/',joiRoom);
+export default chatRoutes;
```

Comment:
you don't need to have this as route just have config of your socket, if you decide to this without route remove this and have it as config of socket